### PR TITLE
PS-547 Add reuseable function for parsing DBC file to Device and DeviceProfile

### PR DESCRIPTION
**Pull request by FelixTing**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## New Dependency Instructions (If applicable)
go.einride.tech/can v0.7.0

### CODE REVIEW

File: central/dbc/compile.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
+// The following code was copied from https://github.com/einride/can-go
```
Comment (cloudxxx8) [inline central/dbc/compile.go:1]: add apache2 license header in any new files

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: central/dbc/compile.go (hunk 1 above)
Comment (FelixTing) [reply central/dbc/compile.go]: added license header

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

**Comment by FelixTing** [comment]
Added DTO validation and return errors in a similar way to the Excel Converter

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**